src/Exception/DuplicateKeyException.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/*
+ * Copyright 2015-2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Exception;
+
+class DuplicateKeyException extends \MongoDB\Driver\Exception\BulkWriteException implements Exception
+{
+    /**
+     * Thrown when inserting a document with a key that is already present in the database.
+     *
+     * @param string $key Key of inserted document
+     * @return self
+     */
+    public static function duplicateKeyInsertion($key)
+    {
+        return new static(sprintf('The key (%s) of this document is already present in the database.', $key));
```

Comment:
Reading this message, it may be ambiguous if the string in parentheses is the field name or duplicate value. Looking at how this is called from `InsertOne.php`, the inserted ID is passed in as `$key`. That may be troublesome for a couple of reasons:

 * While `_id` is certainly a common case of duplicate key errors (especially with non-ObjectID values), it's possible that the exception arose from an index on another field
 * The inserted ID may not be easily represented as as string (e.g. embedded document)

This also potentially hides the server's original error message, which reports the collection name, field name, and offending value (whether it was a document, string, etc.).